shared/processing/dataGetters.ts

```diff
@@ -0,0 +1,319 @@
+            linkedItems,
+        );
+
+        const dataObject: IPathOperation = {
+            ...getSystemProperties(pathOperation),
+            apiReference: processTaxonomyElement(pathOperation.apiReference),
+            category: processLinkedItemsElement(pathOperation.category),
+            codeSamples: processLinkedItemsElement(pathOperation.codeSamples),
+            deprecated: processMultipleChoiceElement(pathOperation.deprecated),
+            description: pathOperation.description.getHtml(),
+            name: pathOperation.name.value,
+            parameters: processLinkedItemsElement(pathOperation.parameters),
+            path: pathOperation.path.value,
+            pathOperation: processTaxonomyElement(pathOperation.pathOperation),
+            requestBody: pathOperation.requestBody.getHtml(),
+            responses: pathOperation.responses.getHtml(),
+            url: pathOperation.url.value,
+        };
+
+        paths.push(getWrappedData(dataObject, pathOperation));
+    });
+
+    return paths;
+};
+
+export const getServersData = (
+    field: RichTextField,
+    dataBlob: IPreprocessedData,
+    linkedItems: ContentItem[],
+): Array<IWrappedItem<IServer>> => {
+    const servers = [];
+
+    field.linkedItemCodenames.map((codename: string) => {
+        const server = getFromLinkedItems<ZapiServer>(codename, linkedItems);
+        const dataObject: IServer = {
+            ...getSystemProperties(server),
+            description: server.description.value,
+            url: server.url.value,
+        };
+
+        servers.push(getWrappedData<IServer>(dataObject, server));
+    });
+
+    return servers;
+};
+
+export const getParametersData = (
+    items: ZapiParameter[],
+    dataBlob: IPreprocessedData,
+    linkedItems: ContentItem[],
+): Array<IWrappedItem<IParameter>> => {
+    const parameters = [];
+
+    items.map((parameter: ZapiParameter) => {
+        processItemsInto<ISchemas>(getSchemaDataFromLinkedItemElement)(parameter.schema, dataBlob, linkedItems);
+        processItemsInto<IDescriptionComponents>(getDescriptionItemsData)(parameter.description, dataBlob, linkedItems);
+
+        const dataObject: IParameter = {
+            ...getSystemProperties(parameter),
+            apiReference: processTaxonomyElement(parameter.apiReference),
+            deprecated: processMultipleChoiceElement(parameter.deprecated),
+            description: parameter.description.getHtml(),
+            example: parameter.example.value,
+            explode: processMultipleChoiceElement(parameter.explode),
+            location: processMultipleChoiceElement(parameter.location),
+            name: parameter.name.value,
+            required: processMultipleChoiceElement(parameter.required),
+            schema: processLinkedItemsElement(parameter.schema),
+            style: processMultipleChoiceElement(parameter.style),
+        };
+
+        parameters.push(getWrappedData<IParameter>(dataObject, parameter));
```

Comment:
Please, get rid of these `push`s - if you really need an variable with results, so be it, but other than that, all these function just transform set of objects of one type to another and a bodyless function is a perfect tool to describe such transformation IMO 😇